### ISSUE TITLE
Page not found for "Using a custom domain" and "Using subroutes" in knative.dev/docs/serving/

### ISSUE THREAD

**Issue by nak3** [knative/docs#4111]
## Expected Behavior

- `Using a custom domain` should point to https://knative.dev/docs/serving/using-a-custom-domain/
- `Using subroutes` should be changed to `Traffic Management` with the link https://knative.dev/docs/developer/serving/traffic-management/

## Actual Behavior

- The link for "Using a custom domain" and "Using subroutes" in https://knative.dev/docs/serving/#configuration-and-networking  gets 404 error.
- One of the link `Using subroutes` page seems not exist anymore as per https://github.com/knative/docs/blob/7ee18955f141dfe01bda511a4460169e954b1101/config/redirects.yml#L5

## Steps to Reproduce the Problem

1. Go to https://knative.dev/docs/serving/using-a-custom-domain/
1. Click "Using a custom domain"  (it will get 404 error)
1. Click "Using subroutes" (it will get 404 error)

**Comment by abrennan89** [comment]
Good catch. Some of the other links on https://knative.dev/docs/serving/using-a-custom-domain/ are also broken so these can be fixed together.
